rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -43,31 +43,56 @@ local function get_current_lb_alg()
   return backend["load-balance"] or DEFAULT_LB_ALG
 end
 
+local function is_sticky(backend)
+  return backend["sessionAffinityConfig"] and backend["sessionAffinityConfig"]["name"] == "cookie"
+end
+
 local function balance()
   local backend = get_current_backend()
   if not backend then
     return nil, nil
   end
-  local lb_alg = get_current_lb_alg()
-  local is_sticky = sticky.is_sticky(backend)
-
-  if is_sticky then
-    local endpoint = sticky.get_endpoint(backend)
-    if endpoint ~= nil then
-      return endpoint.address, endpoint.port
-    end
-    lb_alg = "round_robin"
-  end
 
-  if backend["upstream-hash-by"] then
+  if is_sticky(backend) or backend["upstream-hash-by"] then
     local endpoint = resty_balancer.balance(backend)
     if not endpoint then
       return nil, nil
     end
 
+    if backend["sessionAffinityConfig"]["name"] == "cookie" then
+      local cookie, err = ck:new()
+      if not cookie then
+        ngx.log(ngx.ERR, err)
+      end
+
+      local endpoint_string = endpoint.address .. ":" .. endpoint.port
+      local encrypted
+      if backend["sessionAffinityConfig"]["cookieSessionAffinity"]["hash"] == "sha1" then
```

Comment:
Would be nice to have a helper method here that generates the digested endpoint string rather than doing it over here.